File: app/src/main/java/com/prey/services/MyFirebaseMessagingService.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2018 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.services;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import com.prey.PreyLogger;
import com.prey.beta.actions.PreyBetaController;

public class MyFirebaseMessagingService extends FirebaseMessagingService {

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        PreyLogger.d("FIREBASE onMessageReceived" );
        if (remoteMessage.getData().size() > 0) {
            final String text= remoteMessage.getData().toString();
            PreyLogger.d("FIREBASE data:"+text);
            String cmd=null;
            try {
                cmd = remoteMessage.getData().get("cmd");
            } catch (Exception e) {
                cmd = null;
            }
            PreyLogger.d("FIREBASE cmd:"+cmd);
            PreyBetaController.startPrey(this,cmd);
        }
    }

}

File: app/src/main/java/com/prey/services/MyFirebaseMessagingService.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2018 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.services

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.prey.PreyConfig
import com.prey.PreyLogger
import com.prey.json.parser.JsonCommandDispatcher.getActionsJson
import com.prey.json.parser.JsonCommandDispatcher.getActionsJsonArray

class MyFirebaseMessagingService : FirebaseMessagingService() {

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        // Check if message contains a data payload.
        if (remoteMessage.data.isNotEmpty()) {
            PreyLogger.d("___ Message data payload: ${remoteMessage.data}")
            // Process the data payload here
            val cmd = remoteMessage.data.get("cmd")
            PreyLogger.d("___ cmd: $cmd")
            val config = PreyConfig.getPreyConfig(applicationContext)
            PreyLogger.d("startPrey:" + config.isThisDeviceAlreadyRegisteredWithPrey())
            if (config.isThisDeviceAlreadyRegisteredWithPrey()) {
                config.isRun = true
                if (cmd.isNullOrEmpty()) {
                    getActionsJsonArray(applicationContext)
                } else {
                    getActionsJson(applicationContext, cmd)
                }
            }
        }
    }

    override fun onNewToken(token: String) {
        // Called when a new FCM registration token is generated or updated
        PreyLogger.d("___ Refreshed token: $token")
        // If you want to send this token to your app server, do it here
    }

}